src/main/java/com/whalespottingjava/controllers/ApiControllerAdvice.java

```diff
@@ -0,0 +1,15 @@
+package com.whalespottingjava.controllers;
+
+import jakarta.validation.ConstraintViolationException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+@RestControllerAdvice
+public class ApiControllerAdvice {
+    @ExceptionHandler({ DataIntegrityViolationException.class, ConstraintViolationException.class })
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public String dataIntegrityViolationHandler(Exception ex) {
+        return ex.getLocalizedMessage(); // TODO: 16/10/2023 This reveals implementation details on the database, error messages should be overridden 
```

Comment:
Ditto about this TODO (replace date with ticket number that covers implementing this behaviour)

Reply:
Ticket added